swap_meet/vendor.py

```diff
@@ -1,2 +1,113 @@
+        if both vendors have the specified items in their inventory.
+        """
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        return False
+
+    def swap_first_item(self, other_vendor):
+        """
+        Swaps the first item from this vendor's inventory with the first item
+        from another vendor's inventory.
+        """
+        if len(self.inventory) and len(other_vendor.inventory):
+            my_first_item = self.inventory[0]
+            their_first_item = other_vendor.inventory[0]
+            return self.swap_items(other_vendor, my_first_item, their_first_item)
+
+    def get_by_category(self, category):
+        """
+        Retrieves all items in the inventory that match the given category.
+        """
+        result = []
+        for item in self.inventory:
+            if item.get_category() == category.capitalize():
+                result.append(item)
+        return result
+
+    def get_best_by_category(self, category):
+        """
+        Finds the item with the highest `condition`and matching `category`
+        """
+        items_in_category = self.get_by_category(category)
+        if items_in_category:
+            highest_condition = items_in_category[0]
+            for item in items_in_category:
+                if item.condition > highest_condition.condition:
+                    highest_condition = item
+            return highest_condition
+
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        """
+        Swaps the best item in this vendor's inventory (matching the `their_priority` category)
+        with the best item in the other vendor's inventory (matching the `my_priority` category).
+        """
+        my_best_item = self.get_best_by_category(their_priority)
+        their_best_item = other_vendor.get_best_by_category(my_priority)
+        if my_best_item and their_best_item:
+            return self.swap_items(other_vendor, my_best_item, their_best_item)
+
+    def find_newest(self, vendor):
+        """
+        Finds newest item by most latest date(age) in each inventory
+        """
+        newest_item = vendor.inventory[0]
+        for item in vendor.inventory:
+            if item.age > newest_item.age:
+                newest_item = item
+        return newest_item
+
+    def swap_by_newest(self, other_vendor):
+        """
+        Swaps the newest item from this vendor's inventory with the newest item
+        from another vendor's inventory.
+        """
+        if self.inventory and other_vendor.inventory:
+            my_item = self.find_newest(self)
+            their_item = self.find_newest(other_vendor)
+            return self.swap_items(other_vendor, my_item, their_item)
```

Comment:
Prefer the main logic to not be written in a conditional block. 

We should check for invalid input scenario and do something about it and then lines 111-113 will just run when there is valid input.